lib/gds_api/railtie.rb

```diff
@@ -1,10 +1,15 @@
-require_relative 'middleware/govuk_request_id_sniffer'
+require_relative 'middleware/govuk_header_sniffer'
 
 module GdsApi
   class Railtie < Rails::Railtie
     initializer "gds_api.initialize_govuk_request_id_sniffer" do |app|
-      Rails.logger.info "Using middleware GdsApi::GovukRequestIdSniffer to sniff for GOVUK-Request-Id header"
-      app.middleware.use GdsApi::GovukRequestIdSniffer
+      Rails.logger.info "Using middleware GdsApi::GovukHeaderSniffer to sniff for GOVUK-Request-Id header"
+      app.middleware.use GdsApi::GovukHeaderSniffer, 'HTTP_GOVUK_REQUEST_ID', 'GOVUK-Request-Id'
```

Comment:
I don't think the capitalisation is correct here. Shouldn't it be `Govuk-Request-Id`? Maybe it would be safer if the conversion of the Rack-specification-formatted header name (`HTTP_VARIABLE_NAME`) to the version that the API adapter sets  should happen in `GovukHeaderSniffer` instead of being passed in. Looking at the way headers were being [merged in `JsonClient`](https://github.com/alphagov/gds-api-adapters/pull/317/files#diff-36a775ddd5ecd0c84b6e434ce0b174a1L175) before, it looks like it could simply be converted into a symbolic key, e.g. `header_name.sub(/^HTTP_/, "").downcase.to_sym`.